### DISCUSSION TITLE
Trigger shutter every x distance travelled

### DISCUSSION THREAD

**Discussion by digitalgeotechnics** [gopro/labs#247] [Feature Requests]
It seems the `c` condition can be used to trigger an action at a distance from the cameras startup location.  `c:A` can be used to trigger an action at a defined distance from point `A`. But, I'm trying to get the camera to take a photo every _x_ meters travelled.  The documentation suggests `MFRMA` can be used to set position `A` to the current location but only when there isn't already a value for position `A`.  To make it do what I want I need to be able to continually update position `A`.  That way when, say `c:A10` condition is met I can trigger the shutter, then reset `A` to the current location and repeat.  I guess I'd like to request the ability to overwrite position `A` regardless!

**Comment by dnewman-gpsw** [comment]
The easiest is use a modulus of the distance (radius) from the initial coordinate.  Or another crude solution would be to put the camera to sleep and use a boot command, so that the same distance script runs, with a new A coordinate is loaded on each boot (although the distances moved might need to be greater than 10 meters.)  The script would be something like mP!D>c100!S+!1OR!R  - so on boot the camera set mode Photo (mP), waits for GPS lock (!D), then test is the distance is greater than 100m (>c100) and if true takes a photo and reboots (!S+!1OR) else it repeats without a reboot.

But looking at the code there is a better solution mP!D>c:A10oMFRMA=0,0+!S!R

The code is sees 0,0 coordinates as not set, so if you clear the coordinates, then will be reloaded with with current location on each new true condition.   I just tried it and it works great.

![image](https://user-images.githubusercontent.com/10856084/221934098-f2d3b4fd-14eb-4a4a-9c39-fc98d75134c0.png)

**Reply by digitalgeotechnics** [reply]
That's excellent, just what I was trying to achieve, thanks!

**Comment by dnewman-gpsw** [comment]
This feature is out now in HERO11's 2.10.70 Labs firmware
